common/environment/environment.cpp

```diff
@@ -86,7 +156,30 @@ class CLocalEnvironment : public CInterface, implements IConstEnvironment
     void unlockRemote();
     virtual bool isConstEnvironment() const { return true; }
     virtual void clearCache();
-   
+
+    virtual IConstMachineInfoIterator * getMachineIterator() const;
```

Comment:
This is not used by this PR.
What is it to be used for?

The drop zone iterator methods should be all that Esp etc. need.